### ISSUE TITLE
Invoke inline media player when a playable link is posted in the chatroom

### ISSUE THREAD

**Issue by Edszx** [vector-im/riot-web#4944]
Currently, if we upload a media file that chrome can play (webm, mp4, mp3, ogg, opus) Riot invokes a media player in the message along with the file download link, can we have something like this but for pasted links from the net? (im not even talking about youtube links, but links that directly go to media files like example.com/song.opus)

**Comment by t3chguy** [comment]
This would expose your IP Address to the host and thus is considered unsafe, the only other way would be for Synapse to download it on your behalf which would probably not be instantaneous.

**Comment by Edszx** [comment]
Uploading files from links (possibly with a server-side configurable size limit to prevent abuse) sounds like a very good idea too.

**Comment by t3chguy** [comment]
another issue stems from this *automatic* upload, the file might not be loadable by Javascript due to CORS

**Comment by Edszx** [comment]
In that case, the client couldn't just give us an error message saying that it can't download the file?
Also, with this second idea, im no longer talking about making the cross-upload automatic when the link is sent to the chatroom, now im talking about maybe adding the option to "upload from link" to the current "upload file" button.

**Comment by t3chguy** [comment]
@Edszx I like the upload from link idea, but it might still hit CORS issues

**Comment by t3chguy** [comment]
(unless server accepted a link and downloaded it itself, but that'd require protocol changes)

**Comment by turt2live** [comment]
fwiw, youtube embeds are covered here: https://github.com/vector-im/riot-web/issues/3466

**Comment by Edszx** [comment]
@turt2live Yep, i saw that issue some days ago, but this situation is not the same, nevertheless, both things would be very good additions, i hope the devs could consider them for when they feel like it.

**Comment by ShadowJonathan** [comment]
Related; discord obscures linked media by embedding it as if it was any normal uploaded media.

This isnt maybe an approach Element would have to take (due to security concerns), but at least maybe providing the option to "download the link, and then upload it" for power users, with a similar "simple" mechanism of copying a media link whole-sale into the composer, would be interesting.

**Comment by MadLittleMods** [comment]
Related to https://github.com/vector-im/element-web/issues/437
